cmd/dep/init_test.go

```diff
@@ -39,63 +38,119 @@ func TestIsStdLib(t *testing.T) {
+				"github.com/sdboyer/deptest":    "v0.8.0",                                   // semver
+				"github.com/sdboyer/deptestdos": "a0196baa11ea047dd65037287451d36b861b00ea", // random sha
+			},
+			sourceFiles: map[string]string{
+				"thing.input.go": "foo/thing.go",
+				"bar.input.go":   "foo/bar/bar.go",
+			},
+			goldenManifest: "manifest.golden.json",
+			goldenLock:     "lock.golden.json",
+		},
+
+		// One dependency previously retrieved by version.  Both will show up in lock, but only retrieved one in manifest.
+		{
+			dataRoot: "init/case2",
+			importPaths: map[string]string{
+				"github.com/sdboyer/deptest": "v0.8.0", // semver
+			},
+			sourceFiles: map[string]string{
+				"thing.input.go": "foo/thing.go",
+				"bar.input.go":   "foo/bar/bar.go",
+			},
+			goldenManifest: "manifest.golden.json",
+			goldenLock:     "lock.golden.json",
+		},
+
+		// One dependency previously retrieved by sha.  Both will show up in lock and manifest?
+		{
+			dataRoot: "init/case3",
+			importPaths: map[string]string{
+				"github.com/sdboyer/deptestdos": "a0196baa11ea047dd65037287451d36b861b00ea",
+			},
+			sourceFiles: map[string]string{
+				"thing.input.go": "foo/thing.go",
+				"bar.input.go":   "foo/bar/bar.go",
+			},
+			goldenManifest: "manifest.golden.json",
+			goldenLock:     "lock.golden.json",
+		},
+	}
 
-	h := test.NewHelper(t)
-	defer h.Cleanup()
+	runTest := func(t *testing.T, testCase initTestCase) {
+		test.NeedsExternalNetwork(t)
+		test.NeedsGit(t)
 
-	h.TempDir("src")
-	h.Setenv("GOPATH", h.Path("."))
+		h := test.NewHelper(t)
+		defer h.Cleanup()
 
-	importPaths := map[string]string{
-		"github.com/pkg/errors":      "v0.8.0",                                   // semver
-		"github.com/Sirupsen/logrus": "42b84f9ec624953ecbf81a94feccb3f5935c5edf", // random sha
-	}
+		h.TempDir("src")
+		h.Setenv("GOPATH", h.Path("."))
 
-	// checkout the specified revisions
-	for ip, rev := range importPaths {
-		h.RunGo("get", ip)
-		repoDir := h.Path("src/" + ip)
-		h.RunGit(repoDir, "checkout", rev)
-	}
+		// checkout the specified revisions
+		for ip, rev := range testCase.importPaths {
+			h.RunGo("get", ip)
+			repoDir := h.Path("src/" + ip)
+			h.RunGit(repoDir, "checkout", rev)
+		}
 
-	// Build a fake consumer of these packages.
-	root := "src/github.com/golang/notexist"
-	h.TempCopy(root+"/foo/thing.go", "init/thing.input.go")
-	h.TempCopy(root+"/foo/bar/bar.go", "init/bar.input.go")
-
-	h.Cd(h.Path(root))
-	h.Run("init")
-
-	goldenManifest := "init/manifest.golden.json"
-	wantManifest := h.GetTestFileString(goldenManifest)
-	gotManifest := h.ReadManifest()
-	if wantManifest != gotManifest {
-		if *test.UpdateGolden {
-			if err := h.WriteTestFile(goldenManifest, gotManifest); err != nil {
-				t.Fatal(err)
+		// Build a fake consumer of these packages.
+		root := "src/github.com/golang/notexist"
+		for src, dest := range testCase.sourceFiles {
+			h.TempCopy(root+"/"+dest, testCase.dataRoot+"/"+src)
```

Comment:
Yeah, I debated that move.  Clearly not best practice the way I did it yet I am running on a PC and it is handling the "/" well.

I'll fix them all.  I just wish the name for that function wasn't so long - hurts readability IMO